docs/excel/custom-functions-batching.md

```diff
@@ -0,0 +1,214 @@
+
+The `_makeRemoteRequest` function calls `_fetchFromRemoteService` which as you'll see later is just a mock representing the remote service. This makes it easier to study and run the code in this article. But when you want to use this code for an actual remote service you should make the following changes:
+
+- Decide how to serialize the batch operations over the network. For example, you may want to put the array into a JSON body.
+- Instead of calling `_fetchFromRemoteService` you'll need to make the actual network call to the remote service passing the batch of operations.
+
+## Process the batch call on the remote service
+
+The last step is to handle the batch call in the remote service. The following code sample shows the `_fetchFromRemoteService` function. This function unpacks each operation, performs the specified operation, and returns the results. For learning purposes in this article, the `_fetchFromRemoteService` function is designed to run in your web add-in and mock a remote service. You can add this code to your functions.ts file so that you can study and run all the code in this article without having to set up an actual remote service.
+
+### Add the following _fetchFromRemoteService function to functions.ts
+
+```typescript
+async function _fetchFromRemoteService(
+  requestBatch: Array<{ operation: string, args: any[] }>
+): Promise<IServerResponse[]> {
+  // Simulate a slow network request to the server;
+  await pause(1000);
+
+  return requestBatch.map((request): IServerResponse => {
+    const { operation, args } = request;
+
+    try {
+      if (operation === "div2") {
+        // Divide the first argument by the second argument.
+        return {
+          result: args[0] / args[1]
+        };
+      } else if (operation === "mul2") {
+        // Multiply the arguments for the given entry.
+        const myresult = args[0] * args[1];
+        console.log(myresult);
+        return {
+          result: myresult
+        };
+      } else {
+        return {
+          error: `Operation not supported: ${operation}`
+        };
+      }
+    } catch (error) {
+      return {
+        error: `Operation failed: ${operation}`
+      };
+    }
+  });
+}
+
+function pause(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+```
+
+### Modify _fetchFromRemoteService for your live remote service
+
+To modify the `fetchFromRemoteService` function to run in your live remote service, make the following changes:
+
+- Map from the client network call to this function.
+- Remove the `pause` function which simulates network latency as part of the mock.
+- Modify the function to unpack the actual network package that you defined (for example the batch array of operations may be a JSON body).
+- Modify the function to perform the operations (or call functions that do the operations).
+- Apply an appropriate authentication mechanism. Ensure that only the correct callers can access the function.
```

Comment:
"...correct callers..." > "...authenticated users..." ?

Reply:
@Reezaali this was text approved by Hide, and in a scenario where the server may have automated systems calling it for operations (not just Excel users). All depends on how someone architects their solution I think.